test/conftest.py

```diff
@@ -392,54 +381,36 @@ def pudl_tmpdir(tmp_path_factory):
     return tmpdir
 
 
-@pytest.fixture(scope="session")
-def pudl_output_tmpdir(pudl_tmpdir):
-    tmpdir = pudl_tmpdir / "output"
-    tmpdir.mkdir()
-    return tmpdir
-
-
-@pytest.fixture(scope="session")
-def pudl_input_tmpdir(pudl_tmpdir):
-    tmpdir = pudl_tmpdir / "data"
-    tmpdir.mkdir()
-    return tmpdir
+def pytest_sessionstart(session):
+    """Configures input/output paths for the tests."""
+    # TODO(rousik): Should we be using fixed paths instead
+    # of using tmpdir capabilities here?
+    pudl.workspace.setup.set_path_overrides(
+        input_dir="~/pudl-work/data",
+        output_dir="~/pudl-work/output",
+    )
+    logger.info(f"Starting unit tests with output path {PudlPaths().output_dir}")
+    pudl.workspace.setup.init()
 
 
 @pytest.fixture(scope="session")
-def pudl_input_output_dirs(request, live_dbs, pudl_input_tmpdir, pudl_output_tmpdir):
-    """Determine where the PUDL input/output dirs should be."""
-    input_override = None
-    output_override = None
-
+def pudl_path_setup(request, pudl_tmpdir):
+    """Sets the necessary env variables for the input and output paths."""
     if os.environ.get("GITHUB_ACTIONS", False):
-        # hard-code input dir for CI caching
-        input_override = Path(os.environ["HOME"]) / "pudl-work" / "data"
-        output_override = Path(os.environ["HOME"]) / "pudl-work" / "output"
-    elif request.config.getoption("--tmp-data"):
-        # use tmpdir for inputs if we ask for it
-        input_override = pudl_input_tmpdir
-    if not live_dbs:
-        # use tmpdir for outputs if we haven't passed --live-db
-        output_override = pudl_output_tmpdir
-
-    return {"input_dir": input_override, "output_dir": output_override}
-
-
-@pytest.fixture(scope="session", name="pudl_settings_fixture")
-def pudl_settings_dict(request, pudl_input_output_dirs):  # noqa: C901
-    """Determine some settings (mostly paths) for the test session."""
-    logger.info("setting up the pudl_settings_fixture")
-    pudl_settings = pudl.workspace.setup.get_defaults(**pudl_input_output_dirs)
-    pudl.workspace.setup.init(pudl_settings)
-
-    pudl_settings["sandbox"] = request.config.getoption("--sandbox")
-
-    pretty_settings = json.dumps(
-        {str(k): str(v) for k, v in pudl_settings.items()}, indent=2
-    )
-    logger.info(f"pudl_settings being used: {pretty_settings}")
-    return pudl_settings
+        pudl.workspace.setup.set_path_overrides()
+    else:
+        if request.config.getoption("--tmp-data"):
+            in_tmp = pudl_tmpdir / "data"
+            in_tmp.mkdir()
+            pudl.workspace.setup.set_path_overrides(
+                input_dir=str(Path(in_tmp).resolve()),
+            )
+        if request.config.getoption("--live-dbs"):
+            out_tmp = pudl_tmpdir / "output"
+            out_tmp.mkdir()
+            pudl.workspace.setup.set_path_overrides(
+                output_dir=str(Path(out_tmp).resolve()),
+            )
```

Comment:
This logic seems a little funny to me -- if we've gotten `--live-dbs` then currently we're using the user's already-existing databases rather than building temporary databases from scratch, so the output directory should be the user's default output directory shouldn't it? And not a temporary location? Or are the paths to the user's PUDL DB and FERC DBs being set independently somewhere else?

Reply:
Um, you're right, this doesn't sound right at all.